### PR TITLE
Fixing unchecked and rawtypes warnings.

### PR DESCRIPTION
**Pull request by hakanai**
RTFParserDelegate had a method with raw List on it, which forced us to use List without type parameters, generating a compiler warning. Fixing this one and the rest.

### CODE REVIEW

File: src/com/etranslate/tm/processing/rtf/RTFParser.java [hunk 1]
```diff
@@ -174,7 +174,7 @@ private static final String getJavaEncoding(int rtfCodepage) {
   // the current number of bytes to skip after a unicode character
   private Integer _currentSkipState = DEFAULT_SKIP_STATE;
   // a stack of skip states for bytes following a unicode character
-  private final Stack _ucSkipStates = new Stack();
+  private final Stack<Integer> _ucSkipStates = new Stack<>();
```
Comment (tmyroadctfig) [inline src/com/etranslate/tm/processing/rtf/RTFParser.java]: This file is generated from RTFParser.jj, so you need to put the fixes in there.

Comment (hakanai) [reply src/com/etranslate/tm/processing/rtf/RTFParser.java]: Tried that.

```
javacc:
     [java] Java Compiler Compiler Version 4.0 (Parser Generator)
     [java] (type "javacc" with no arguments for help)
     [java] Reading from file RTFParser.jj . . .
     [java] org.javacc.parser.ParseException: Encountered ">" at line 206, column 58.
     [java] Was expecting one of:
     [java]     "boolean" ...
     [java]     "byte" ...
     [java]     "char" ...
     [java]     "double" ...
     [java]     "float" ...
     [java]     "int" ...
     [java]     "long" ...
     [java]     "short" ...
     [java]     "?" ...
     [java]     <IDENTIFIER> ...
     [java]     
     [java] Detected 1 errors and 0 warnings.
```

Comment (hakanai) [reply src/com/etranslate/tm/processing/rtf/RTFParser.java]: So about this one, is there an escape character to dodge this sort of error?

Comment (tmyroadctfig) [reply src/com/etranslate/tm/processing/rtf/RTFParser.java]: I had a quick check last time, and I think the version of javacc needs to be bumped to 5.0 or later.
https://java.net/projects/javacc/lists/users/archive/2012-05/message/12

**Pull request merged**